app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -26,12 +28,60 @@ def get_middleware_servers
           @ems.eaps(feed).map do |eap|
             @eaps << eap
             server = parse_middleware_server(eap)
+
+            machine_id = @ems.machine_id(eap.feed)
+            host_instance = find_host_by_bios_uuid(machine_id)
+            if host_instance.nil?
+              host_instance = find_host_by_bios_uuid(alternate_machine_id(machine_id))
+            end
+
+            unless host_instance.nil?
+              server[:lives_on_id] = host_instance.try(:id)
+              server[:lives_on_type] = host_instance.try(:type)
```

Comment:
No idea. I used this as reference: 
https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb#L172-L184

I didn't make sense to me as well, but I thought it was to handle some corner case.